src/core/ucc_coll.c

```diff
@@ -297,18 +329,20 @@ UCC_CORE_PROFILE_FUNC(ucc_status_t, ucc_collective_post, (request),
     ucc_status_t status;
 
     if (ucc_global_config.coll_trace.log_level >= UCC_LOG_LEVEL_DEBUG) {
-        ucc_rank_t rank = task->team->params.team->rank;
-        if (ucc_global_config.coll_trace.log_level == UCC_LOG_LEVEL_DEBUG) {
-            if (rank == 0) {
+        if (task->team) {
+            ucc_rank_t rank = task->team->params.team->rank;
+            if (ucc_global_config.coll_trace.log_level == UCC_LOG_LEVEL_DEBUG) {
```

Comment:
perhaps remove second if (line 335) and else (line 340) and instead change to single if:
if (ucc_global_config.coll_trace.log_level != UCC_LOG_LEVEL_DEBUG || rank == 0) {
                ucc_log_component_collective_trace(
                    ucc_global_config.coll_trace.log_level,
                    "coll post: rank %d req %p, seq_num %u", rank, task,
                    task->seq_num);
}

Reply:
it will be different in that case since rank 0 will print line twice